src/utils/time.ts

```diff
@@ -53,57 +86,23 @@ export const secondsToDuration = (totalSeconds: number): Duration => {
 export const formatSecondsAsDuration = (seconds: number): string =>
   formatDuration(secondsToDuration(seconds));
 
-const formatMeasurementInString = (value: number, measureUnit: string) => {
-  const baseFormatted = `${value} ${measureUnit}`;
-
-  if (value === 0) return "";
-  if (value === 1) return baseFormatted;
-  if (value > 1) return `${baseFormatted}s`;
-};
-
-// @debt quality test this
 export const getTimeBeforeParty = (startUtcSeconds?: number) => {
   if (startUtcSeconds === undefined) return "???";
 
-  const secondsBeforeParty =
-    startUtcSeconds - Date.now() / ONE_SECOND_IN_MILLISECONDS;
-
-  if (secondsBeforeParty < 0) {
-    return 0;
-  }
+  const eventStartDate = fromUnixTime(startUtcSeconds);
+  const now = Date.now();
 
-  const numberOfCompleteDaysBeforeParty = Math.floor(
-    secondsBeforeParty / ONE_DAY_IN_SECONDS
-  );
-
-  const numberOfCompleteHours = Math.floor(
-    (secondsBeforeParty % ONE_DAY_IN_SECONDS) / ONE_HOUR_IN_SECONDS
-  );
+  if (isBefore(eventStartDate, now)) return 0;
 
-  const numberOfMinutes = Math.floor(
-    (secondsBeforeParty % ONE_HOUR_IN_SECONDS) / ONE_MINUTE_IN_SECONDS
-  );
-
-  const numberOfDaysInString = formatMeasurementInString(
-    numberOfCompleteDaysBeforeParty,
-    "day"
-  );
-  const numberOfHoursInString = formatMeasurementInString(
-    numberOfCompleteHours,
-    "hour"
+  return formatDuration(
+    intervalToDuration({
+      start: Date.now(),
```

Comment:
Since you already set `const now = Date.now()` above in this same function, it would be good to  also use `now` here, as otherwise it will be returning a potentially different time.

Reply:
used `now`: https://github.com/sparkletown/sparkle/pull/1302/commits/61c8d31c7eecfeae9b1debbdc10b22f31a630afe